src/main/kotlin/org/rust/ide/annotator/RustHighlightingAnnotator.kt

```diff
@@ -12,7 +12,11 @@ import org.rust.lang.core.psi.util.elementType
 
 // Highlighting logic here should be kept in sync with tags in RustColorSettingsPage
 class RustHighlightingAnnotator : Annotator {
-    override fun annotate(element: PsiElement, holder: AnnotationHolder) = element.accept(object : RustElementVisitor() {
+
+    override fun annotate(element: PsiElement, _holder: AnnotationHolder) {
+        element.accept(object : RustElementVisitor() {
+        var holder = wrap(_holder)
```

Comment:
How should that work for resolving compound elements, is it still fine to skip? 

It would be nice to instead be able to produce a map of leaf elements to colors, and then have a visitor resolve leaf elements.. but does that actually make sense to do?